### PR TITLE
I6310 adding flow to btn component

### LINKED ISSUE #6310: Add flow types to Button component

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #6310

### CODE REVIEW

File: src/ui/components/Button/index.js [hunk 1]
```diff
@@ -51,7 +58,9 @@ export default class Button extends React.Component {
       to,
       ...rest
     } = this.props;
-    const props = { ...rest };
+    const props: Object = { ...rest };
```
Comment (rebmullin) [inline src/ui/components/Button/index.js]: I am not sure if Object is ok here or if I should create up something like "internalProps". Initially I went that route but it wasn't working so with buttonType. if that's preferred tho, I will look into that more..

File: src/ui/components/Button/index.js [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
 import * as React from 'react';
-import PropTypes from 'prop-types';
 
 import Link from 'amo/components/Link';
 import log from 'core/logger';
 
 import './styles.scss';
 
+type Props = {|
+  buttonType: string | 'none',
```
Comment (willdurand) [inline src/ui/components/Button/index.js]: We should use a better type that is the intersection of all possible values of `BUTTON_TYPES`. This must be done manually though...

```
type ButtonType =
  | 'neutual'
  | 'action'
  | ...;
```

File: src/ui/components/Button/index.js (hunk 1 above)
Comment (willdurand) [reply src/ui/components/Button/index.js]: We can use `$Rest` I think.

File: src/ui/components/Button/index.js (hunk 2 above)
Comment (willdurand) [reply src/ui/components/Button/index.js]: Yes, it means there are other components to fix. See the diff I pasted above, that should help you (hopefully)!

Comment (rebmullin) [reply src/ui/components/Button/index.js]: oh I see in your recommended patch how you we are importing this to each file and fixing this 👍

File: src/ui/components/Button/index.js [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+  | 'cancel'
+  | 'confirm'
+  | 'alert'
+  | 'none';
+
+type Props = {|
+  buttonType: ButtonType,
+  children?: React.Node,
+  className?: string,
+  disabled: boolean | null,
```
Comment (willdurand) [inline src/ui/components/Button/index.js]: I think we should not have that, it should be `boolean` only

File: src/ui/components/Button/index.js [hunk 4]
```diff
@@ -1,13 +1,40 @@
...
+  externalDark?: boolean,
+  href?: string,
+  micro: boolean,
+  name?: number,
+  noLink: boolean,
+  onClick?: Function | null,
+  puffy: boolean,
+  to?: string | Object,
```
Comment (willdurand) [inline src/ui/components/Button/index.js]: Can you add a `// TODO: make a better Object type` here?

File: src/ui/components/Button/index.js (hunk 3 above)
Comment (willdurand) [reply src/ui/components/Button/index.js]: It requires a change in the `UserProfile` (or `UserProfileEdit`) component.

Comment (rebmullin) [reply src/ui/components/Button/index.js]: ah! I missed this above. I'll update..

**Pull request merged**